sktime/alignment/dtw_python.py

```diff
@@ -0,0 +1,337 @@
+
+_check_soft_dependencies("dtw")
+
+
+class AlignerDTW(BaseAligner):
+    """Aligner interface for dtw-python.
+
+    Behaviour: computes the full alignment between X[0] and X[1]
+        assumes pairwise alignment (only two series) and univariate
+        if multivariate sequences are passed:
+            alignment is computed on univariate series with variable_to_align;
+            if this is not set, defaults to the first variable of X is used
+        raises an error if variable_to_align is not present in X[0] or X[1]
+
+    Parameters
+    ----------
+    dist_method: string, pointwise (local) distance function to use
+        one of the functions in `scipy.spatial.distance.cdist`
+        default = 'euclidean'
+    step_pattern: a stepPattern object describing the local warping steps
+        one of: 'symmetric1', 'symmetric2' (default), 'asymmetric',
+                and dozens of other more non-standard step patterns;
+                list can be displayed by calling help(stepPattern) in dtw
+    window_type: string, the chosen windowing function
+        "none", "itakura", "sakoechiba", or "slantedband"
+            "none" (default) - no windowing
+            "sakoechiba" - a band around main diagonal
+            "slantedband" - a band around slanted diagonal
+            "itakura" - Itakura parallelogram
+    open_begin, open_end: boolean; whether to perform open-ended alignments
+        open_begin = whether alignment open ended at start (low index)
+        open_end = whether alignment open ended at end (high index)
+    variable_to_align: string; which variable to use for univariate alignment
+        default: first variable in X as passed to fit
+    """
+
+    _tags = {
+        "capability:multiple-alignment": False,  # can align more than two sequences?
+        "capability:distance": True,  # does compute/return overall distance?
+        "capability:distance-matrix": True,  # does compute/return distance matrix?
+    }
+
+    def __init__(
+        self,
+        dist_method="euclidean",
+        step_pattern="symmetric2",
+        window_type="none",
+        open_begin=False,
+        open_end=False,
+        variable_to_align=None,
+    ):
+        """Construct instance."""
+        super(AlignerDTWdist, self).__init__()
+
+        self.dist_method = dist_method
+        self.step_pattern = step_pattern
+        self.window_type = window_type
+        self.open_begin = open_begin
+        self.open_end = open_end
+        self.variable_to_align = variable_to_align
+
+    def _fit(self, X, Z=None):
+        """Fit alignment given series/sequences to align.
+
+            core logic
+
+        Parameters
+        ----------
+        X: list of pd.DataFrame (sequence) of length n - panel of series to align
+        Z: pd.DataFrame with n rows, optional; metadata, row correspond to indices of X
+
+        Writes to self
+        --------------
+        alignment - computed alignment from dtw package
```

Comment:
```suggestion
        alignment - computed alignment from dtw package
```
might be worth saying something about the (machine) type of `alignment`